File: Patrones/Estructurales/Decorator/Java/src/DecoradorBebida.java
public abstract class DecoradorBebida implements Bebida {
    protected Bebida bebidaDecorada;

    public DecoradorBebida(Bebida bebida) {
        this.bebidaDecorada = bebida;
    }

    @Override
    public String getDescripcion() {
        return bebidaDecorada.getDescripcion();
    }

    @Override
    public double getCosto() {
        return bebidaDecorada.getCosto();
    }
}

File: Patrones/Estructurales/Decorator/Python/DecoradorBebida.py
from Bebida import Bebida

# Clase abstracta DecoradorBebida que implementa Bebida
class DecoradorBebida(Bebida):

    def __init__(self, bebida_decorada):
        self._bebida_decorada = bebida_decorada

    def get_descripcion(self):
        return self._bebida_decorada.get_descripcion()

    def get_costo(self):
        return self._bebida_decorada.get_costo()
